code/modules/xenoarchaeology/xenoartifact.dm

```diff
@@ -0,0 +1,470 @@
+			if(!xenop.price)
+				xenop.price = pick(300, 500, 800) 
+			malfunction_mod = 1
+
+		if(XENOA_BANANIUM)
+			name = "bananium [name]"
+			generate_traits(list(/datum/xenoartifact_trait/major/sing))
+			if(!xenop.price)
+				xenop.price = pick(500, 800, 1000) 
+			malfunction_mod = 0.25
+
+	//Initialize traits that require that.
+	for(var/datum/xenoartifact_trait/t as() in traits)
+		t.on_init(src)
+
+	//Random sprite process, I'd like to maybe revisit this, make it a function. probably don't
+	icon_state = null
+	if(!(icon_state))
+		var/holdthisplease = rand(1, 3)
+		icon_state = "IB[holdthisplease]"//base
+		generate_icon(icon, "IBL[holdthisplease]", material)
+	if(prob(50) || icon_slots[1])//Top
+		if(!(icon_slots[1])) //Some traits can set this too, it will be set to a code that looks like 901, 908, 905 ect.
+			icon_slots[1] = rand(1, 2)
+		generate_icon(icon, "ITP[icon_slots[1]]")
+		generate_icon(icon, "ITPL[icon_slots[1]]", material)
+
+	if(prob(30) || icon_slots[3])//Left
+		if(!(icon_slots[3]))
+			icon_slots[3] = rand(1, 2)
+		generate_icon(icon, "IL[icon_slots[3]]")
+		generate_icon(icon, "ILL[icon_slots[3]]", material)
+
+	if(prob(50)  || icon_slots[4])//Right
+		if(!(icon_slots[4]))
+			icon_slots[4] = rand(1, 2)
+		generate_icon(icon, "IR[icon_slots[4]]")
+		generate_icon(icon, "IRL[icon_slots[4]]", material)
+
+	if(prob(30) || icon_slots[2])//Bottom
+		if(!(icon_slots[2]))
+			icon_slots[2] = rand(1, 2)
+		generate_icon(icon, "IBTM[icon_slots[2]]")
+		generate_icon(icon, "IBTML[icon_slots[2]]", material)
+
+/obj/item/xenoartifact/CanAllowThrough(atom/movable/mover, turf/target) //tweedle dee, density feature
+	if(get_trait(/datum/xenoartifact_trait/minor/dense))
+		return FALSE
+	return ..()
+
+/obj/item/xenoartifact/attack_hand(mob/user) //tweedle dum, density feature
+	if(get_trait(/datum/xenoartifact_trait/minor/dense))
+		if(process_type == PROCESS_TYPE_LIT) //Snuff out candle
+			to_chat(user, "<span class='notice'>You snuff out [name]</span>")
+			process_type = null
+			return FALSE
+			
+		if(user.a_intent != INTENT_GRAB)
+			SEND_SIGNAL(src, XENOA_INTERACT, null, user, user) //Calling the regular attack_hand signal causes feature issues, like picking up the artifact.
+		else if(touch_desc?.on_touch(src, user) && user.can_see_reagents())
+			balloon_alert(user, (initial(touch_desc.desc) ? initial(touch_desc.desc) : initial(touch_desc.label_name)), material)
+		return FALSE
+	..()
+
+/obj/item/xenoartifact/examine(mob/living/carbon/user)
+	. = ..()	
+	if(isobserver(user))
+		to_chat(user, "<span class='notice'>[special_desc]</span>")
+		for(var/datum/xenoartifact_trait/t as() in traits)
+			to_chat(user, "<span class='notice'>[t?.desc ? t.desc : t.label_name]</span>")
+			to_chat("test")
+	else if(iscarbon(user) && user.can_see_reagents()) //Not checking carbon throws a runtime concerning observers
+		to_chat(user, "<span class='notice'>[special_desc]</span>")
+	return (label_desc ? . + label_desc : .)
```

Comment:
```suggestion
	return "[.][label_desc]"
```
I think this should work, as if label_desc is null, it will be replaced with nothing.

![image](https://user-images.githubusercontent.com/26465327/178124523-a6464606-6f39-481c-8094-66e6c83d6da7.png)